### ISSUE TITLE
Vivaldi does not enable "Hardware acceleration" and WebGL.

### ISSUE THREAD

**Issue by Sumi-Sumi** [NixOS/nixpkgs#204109]
### Describe the bug
Vivaldi does not enable "Hardware acceleration" and WebGL.

### Steps To Reproduce
1. Build vivaldi & Launch vivaldi

Additional-1 steps to reproduce the behavaior:
1. Build vivaldi
2. Launch vivaldi with options.
```
$ vivaldi --ignore-gpu-blocklist --enable-gpu-rasterization --enable-zero-copy
or
$ vivaldi --ignore-gpu-blocklist --enable-gpu-rasterization --enable-zero-copy --use-gl={egl or wayland or desktop}
```

Additional-2 steps to reproduce the behavaior:
1. Override this.
``` 
  (final: prev: {
    vivaldi = prev.vivaldi.overrideAttrs (old: {
     # Add dependence
      buildInputs = old.buildInputs ++ [
        prev.wayland
      ];
      nativeBuildInputs = old.nativeBuildInputs ++ [
        prev.wayland
      ];
      installPhase = ''
          runHook preInstall
          mkdir -p "$out"
          cp -r opt "$out"
          mkdir "$out/bin"
          ln -s "$out/opt/vivaldi/vivaldi" "$out/bin/vivaldi"
          mkdir -p "$out/share"
          cp -r usr/share/{applications,xfce4} "$out"/share
          substituteInPlace "$out"/share/applications/*.desktop \
            --replace /usr/bin/vivaldi "$out"/bin/vivaldi
          substituteInPlace "$out"/share/applications/*.desktop \
            --replace vivaldi-stable vivaldi
          local d
          # Link libwayland-client.so and libwayland-client.so.0 to vivaldi
          ln -s "${prev.wayland}/lib/libwayland-client.so" $out/opt/vivaldi/lib/libwayland-client.so
          ln -s "${prev.wayland}/lib/libwayland-client.so.0" $out/opt/vivaldi/lib/libwayland-client.so.0
          ln -s "${prev.wayland}/lib/libwayland-client.so" $out/opt/vivaldi/libwayland-client.so
          ln -s "${prev.wayland}/lib/libwayland-client.so.0" $out/opt/vivaldi/libwayland-client.so.0
          for d in 16 22 24 32 48 64 128 256; do
            mkdir -p "$out"/share/icons/hicolor/''${d}x''${d}/apps
            ln -s \
              "$out"/opt/vivaldi/product_logo_''${d}.png \
              "$out"/share/icons/hicolor/''${d}x''${d}/apps/vivaldi.png
          done
          wrapProgram "$out/bin/vivaldi" \
            --add-flags ${prev.lib.escapeShellArg ""} \
            --suffix XDG_DATA_DIRS : ${prev.gtk3}/share/gsettings-schemas/${prev.gtk3.name}/ \
            --suffix LD_LIBRARY_PATH : ${old.libPath}
        '' + ''
          runHook postInstall
        '';
   });
```
2. Build vivaldi & Launch vivaldi

### Expected behavior
Enable "Hardware acceleration" and WebGL

### Screenshots
- `--use-gl=egl` 
![image](https://user-images.githubusercontent.com/24451978/205306120-5e07c2dd-ecc8-43c5-a800-fd5473fcb730.png)

- `--use-gl=wayland`
![image](https://user-images.githubusercontent.com/24451978/205305941-329472d6-53d2-4688-8d5f-c13c904a1fe9.png)

- `--use-gl=desktop`
![image](https://user-images.githubusercontent.com/24451978/205306275-d2ae790e-1d44-434f-8613-9a10d7b36c85.png)

- Check `vivaldi:gpu`
![image](https://user-images.githubusercontent.com/24451978/205305814-cc3d43cb-d3a3-4e38-b2bb-dd3c85a4ff6c.png)

### Additional context
Chromium and brave enable "Hardware acceleration" and WebGL. (2022/12/02)
Running environment is Xorg.

### Notify maintainers

@otwieracz @badmutex

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.0.9, NixOS, 23.05 (Stoat), 23.05.20221127.a115bb9`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(sumi): `""`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by MasterMind2k** [comment]
Same issue is present in NixOS 22.11.

**Comment by futile** [comment]
Same problem with vivaldi from `nixpkgs-unstable`, with rest of my system based on `nixos-22.11`. Pretty major, prevents playing Youtube video for me.

**Comment by MasterMind2k** [comment]
Interesting, running vivaldi with `--enable-gl=desktop` worked for me.

```
user@system ~ $ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.80, NixOS, 22.11 (Raccoon), 22.11.466.596a8e828c5`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(root): `"nixos-22.11"`
 - channels(user): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by Sumi-Sumi** [comment]
I tried your solution, but vivaldi have not been work yet in my environment.
I have same issue related to wayland.
Does it depend on the kind of GPU?
I runned under NVIDIA or AMD GPU and used `opengl.extraPackages = with pkgs; [ libvdpau-va-gl vaapiVdpau rocm-opencl-icd # For AMD amdvlk # For AMD ]`

**Comment by MasterMind2k** [comment]
For system packages, I also included `vivaldi-ffmpeg-codecs`. I don't use `opengl.extraPackages`.

**Comment by Sumi-Sumi** [comment]
This issue has been resolved in vivaldi 5.6 and I will close the issue.
